airbyte-integrations/connectors/source-intercom/source_intercom/schemas/conversations.json

```diff
@@ -119,7 +119,9 @@
         }
       }
     },
-    "priority": {
+    "custom_attributes": {
+      "type": ["null", "object"]
```

Comment:
Would we want to optionally specify in the connector spec (configuration) the custom JSON Schema for that field though? If left empty, it could default to object.
 
But if the user is able to describe what kind of data is being recorded there, then normalization would be able to extra more structured data out...

Is that worth exposing to the connector configuration?

Reply:
The ultimate goal is to bring in in all of the `custom_attributes` that are configured in Intercom (similar to how the Salesforce connector works to bring in all custom fields). 

The data is stored as "key:value" in Intercom ([source](https://developers.intercom.com/intercom-api-reference/reference/conversation-model#:~:text=will%20return%20null.-,custom_attributes,different%20custom%20attributes%20associated%20to%20the%20conversation%20as%20key%2Dvalue%20pairs.,-open)), so I don't think we need to allow for JSON in the config since it is relatively simple.

Let me know if I am misunderstanding here though!

Reply:
Is it possible for you to share an example of a record from intercom that includes data in the "custom_attributes" column?

Here in the test / expected results from this PR, it's left empty:

```
"custom_attributes": {},
```

So it's not super insightful on how it'd be presented and used by users afterwards

Reply:
Yes - we don't have any custom attributes created on our Test Intercom account, so that is why they are empty in the example and integration test.

Here is an example result from the prod Intercom account: 
```
"custom_attributes":{"Needed Escalation":false,"Follow Up Required":false,"Chat Closed Reason":"Alpha Connector Issue"}
```

Reply:
It ends up in the database as a json column where you can extract the keys/values as follows:
```
select 
    json_extract_scalar(custom_attributes, "$.Needed Escalation") as needed_escalation,
    json_extract_scalar(custom_attributes, "$.Follow Up Required") as follow_up_required,
    json_extract_scalar(custom_attributes, "$.Chat Closed Reason") as chat_closed_reason
from conversations
```

Reply:
yes, so when running the `discover` command, if the catalog would be able to contain:

```
"custom_attributes": {
      "type": ["null", "object"]
      "properties": {
            "Needed Escalation": {
                    "type": ["null", "string"]
            },
            "Follow Up Required": {
                    "type": ["null", "string"]
            },
            "Chat Closed Reason": {
                    "type": ["null", "string"]
            }
      } 
}
```

then normalization would be able to generate the json_extract lines that you linked automatically too

Reply:
The logic for custom_attributes should be applied, we need to nest all keys into "key" and all the value into "value, than modify the json_schema with this change and it will work correctly for all possible key:value pairs custom_attributes holds.

We can do this now or later on, if we move with the change like in this PR - it's totally ok, but there is a right way to do it, so probably we will make it in the future updates.